2_process/src/process_and_style.R

```diff
@@ -1,19 +1,11 @@
-process_data <- function(nwis_data){
+clean_data <- function(nwis_data, nwis_info_filename){
+  site_info <- read_csv(nwis_info_filename, show_col_types = FALSE)
   nwis_data_clean <- rename(nwis_data, water_temperature = X_00010_00000) %>% 
     select(-agency_cd, -X_00010_00000_cd, -tz_cd)
-  
-  return(nwis_data_clean)
-}
-
-annotate_data <- function(site_data_clean, site_filename){
-  site_info <- read_csv(site_filename)
-  annotated_data <- left_join(site_data_clean, site_info, by = "site_no") %>% 
+  nwis_data_annotated <- left_join(nwis_data_clean, site_info, by = "site_no") %>% 
     select(station_name = station_nm, site_no, dateTime, water_temperature, latitude = dec_lat_va, longitude = dec_long_va)
+  mutate(nwis_data_annotated, station_name = as.factor(station_name))
   
-  return(annotated_data)
+  return(nwis_data_annotated)
 }
```

Comment:
This all looks great, and no need for changes here, but just wanted to mention that you could chain all of this together, rather than having a number of intermediate objects. The whole [pipe](https://style.tidyverse.org/pipes.html) (`%>%`) syntax was new to me coming to R from python, so I figure it may be to you as well. So for example, we could reduce this to fewer lines of codes and intermediate objects as follows:

```r
clean_data <- function(nwis_data, nwis_info_filename) {
  site_info <- read_csv(nwis_info_filename)
  nwis_data_clean <- nwis_data %>%
      rename(water_temperature = X_00010_00000) %>% 
      select(-agency_cd, -X_00010_00000_cd, -tz_cd) %>%
      left_join(site_info, by = "site_no") %>% 
      select(station_name = station_nm, site_no, dateTime, water_temperature, latitude = dec_lat_va, longitude = dec_long_va) %>%
      mutate(station_name = as.factor(station_name))
}
```

Reply:
Ooh, that looks really clean and readable. Thanks for laying this out for me - R's pipe syntax is new to me. It looks a lot like the pipe command in *nix, functionally.